src/matchers/excel.js

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+const { OneDrive } = require('@adobe/helix-onedrive-support');
+const XLSX = require('xlsx');
+
+async function extract(url, params, log = console) {
+  const {
+    AZURE_WORD2MD_CLIENT_ID: clientId,
+    AZURE_HELIX_USER: username,
+    AZURE_HELIX_PASSWORD: password,
+  } = params;
+
+  try {
+    const drive = new OneDrive({
+      clientId,
+      username,
+      password,
+      log,
+    });
+
+    const driveItem = await drive.getDriveItemFromShareLink(url);
+    const xlsxbuffer = await drive.downloadDriveItem(driveItem);
+
+    const workbook = XLSX.read(xlsxbuffer, { type: 'buffer' });
+    const worksheet1 = workbook.Sheets[workbook.SheetNames[0]];
+    const json = XLSX.utils.sheet_to_json(worksheet1);
+
+    return {
+      statusCode: 200,
+      headers: {
+        'Content-Type': 'application/json',
+        'Cache-Control': 'max-age=600',
+      },
+      body: json,
+    };
+  } catch (e) {
+    log.error(e.message);
+    return {
+      statusCode: 500,
+      headers: {
+        'Content-Type': 'application/json',
+        'Cache-Control': 'max-age=600',
+      },
+      body: [],
+    };
+  }
+}
+
+
+module.exports = {
+  required: ['share', 'AZURE_WORD2MD_CLIENT_ID', 'AZURE_HELIX_USER', 'AZURE_HELIX_PASSWORD'],
+  pattern: (url) => /^https:\/\/.*\.sharepoint\.com\/sites\/.*\/_layouts\/15\/guestaccess\.aspx/.test(url),
```

Comment:
I don't think this pattern is very stable....

Reply:
What pattern would you suggest?

Reply:
maybe just `*.sharepoint.com`

Reply:
eg, when I create a sharelink it looks like:

```
https://adobe-my.sharepoint.com/personal/tripod_adobe_com/_layouts/15/guestaccess.aspx?guestaccesstoken=o6QWpgiFUL4qgZYF%2B4Ii%2FExwITlzrDZojOY8V6jXPNI%3D&docid=2_0422f18e696e74ecab0faa476157fd893&rev=1&e=bBt0Oi
```

Reply:
or:

```
https://adobe-my.sharepoint.com/personal/tripod_adobe_com/Documents/helix-content/Book.xlsx?d=w422f18e696e74ecab0faa476157fd893&csf=1&e=EwJ9lg
```